android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/ExtensionProcessDisabledPopupFeature.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.webextensions
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+
+/**
+ * Feature implementation that shows a popup when the extensions process spawning has been
+ * disabled due to the restart threshold being met.
+ *
+ * @property store the application's [BrowserStore].
+ * @property onShowExtensionProcessDisabledPopup a callback invoked when the application should open a
+ * popup.
+ */
+open class ExtensionProcessDisabledPopupFeature(
```

Comment:
@zmckenney @Amejia481 

Drive-by: "feature" classes and components are usually the highest level of a component. This starts [a new pattern](https://searchfox.org/mozilla-mobile/search?q=open+fun+%5BA-Za-z%5D%2BFeature&path=&case=false&regexp=true) that feature classes are extendable and also not in a feature component.

A possible solution is to rename this to something else. Maybe `ExtensionProcessDisabledPopupObserver` or similar?

Reply:
ya, I also had this concern and didn't like calling it a "feature" and originally it was almost directly copied from the [WebExtensionPopupFeature](https://github.com/mozilla-mobile/firefox-android/blob/683843f83e9fdfd5ff00b32a175824ff19ae298b/android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionPopupFeature.kt#L4). Especially now that it's `open` I'm in favor of updating the naming of this, though, and will do that in a separate PR.

Reply:
WebExtensionPopupFeature is one of those that should be renamed too. 🙂 

These seem more like "support" classes than features themselves.

Reply:
@zmckenney please file a bug and attach a PR to fix the naming. It isn't really high priority or anything but let's get that fixed while it's still fresh in our mind.

Reply:
👍🏽